### PR TITLE
Cypress integration migrated to 11+

### LINKED ISSUE #116: Migrate to Cypress 11
### Is your enhancement related to a problem? Please describe.

Cypress 11 has been released, it has a new configuration structure, and the test suite needs to be migrated.

https://docs.cypress.io/guides/references/migration-guide

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jayedul**
### Description of the Change
Cypress integration migrated to 11+

Closes #116 

### How to test the Change
- Install NPM dependencies
- Run `npm run cypress:run` to test without GUI
- Run `npm run cypress:open` to test with GUI
- Check if tests pass successfully

### Changelog Entry

> Changed - Cypress integration migrated to 11+
### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by jeffpaul** [comment]
@jayedul are there any additional work needed here or can this PR come out of draft for review?

**Comment by jayedul** [comment]
@jeffpaul 
This pull request is ready. However the Github action is failing due to WP ENV setup issue. I just checked it happens in all other opened PR also.

As it looks like, WP minimum version attempts to run on updated PHP environment. That's why we see the error `PHP Fatal error:  Uncaught Error: Call to undefined function get_magic_quotes_gpc() in /var/www/html/wp-includes/load.php:651`

File: package.json [hunk 1]
```diff
@@ -9,13 +9,13 @@
...
-    "cypress": "^9.5.2",
+    "cypress": "^11.2.0",
     "node-sass": "^7.0.3",
     "node-wp-i18n": "^1.2.6"
   },
   "scripts": {
-    "cypress:open": "cypress open --config-file tests/cypress/config.json",
-    "cypress:run": "cypress run --config-file tests/cypress/config.json",
+    "cypress:open": "cypress open --config-file tests/cypress/config.config.js",
+    "cypress:run": "cypress run --config-file tests/cypress/config.config.js",
```
Comment (Sidsector9) [inline package.json:18]: @jayedul can you confirm if the file name is correct?

File: tests/cypress/config.config.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const { defineConfig } = require('cypress')
```
Comment (Sidsector9) [inline tests/cypress/config.config.js:1]: Can we rename this to `config.js`

**Review by Sidsector9 — CHANGES_REQUESTED** [review]

**Comment by Sidsector9** [comment]
@jayedul I have synced the latest `develop` with this branch, all expect `trunk` is passing. Can you please check?

File: package.json (hunk 1 above)
Comment (jayedul) [reply package.json:18]: Yes, names are correct as long as the file exists in `./tests/cypress/` directory. Cypress 11+ moved to `.js` from `.json` for configuration.

File: tests/cypress/config.config.js (hunk 2 above)
Comment (jayedul) [reply tests/cypress/config.config.js:1]: We can. Then it should be changed in `package.json` file too. That's all, I believe. I didn't change as it was automatically migrated from `config.json` by cypress.

**Comment by jayedul** [comment]
@Sidsector9 
Now all the tests pass successfully.

**Review by Sidsector9 — APPROVED** [review]
Thank you very much for working on this @jayedul 
LGTM 👍

**Pull request merged**